inst/templates/ad_advs.R

```diff
@@ -0,0 +1,271 @@
+# Name: ADVS
+#
+# Label: Vital Signs Analysis dataset
+#
+# Input: adsl, vs
+
+
+# Attach/load required packages ----
+library(admiral)
+library(admiralmetabolic)
+library(tibble)
+library(dplyr)
+library(stringr)
+
+
+# Define project options/variables ----
+# Use the admiral option functionality to store subject key variables in one
+# place
+set_admiral_options(subject_keys = exprs(STUDYID, USUBJID))
+
+# Store ADSL variables required for derivations as an R object, enabling
+# simplified usage throughout the program
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01P, TRT01A)
+
+
+# Read in data ----
+# See the "Read in Data" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#readdata)
+
+# Read data
+vs_metabolic <- admiralmetabolic::vs_metabolic
+adsl <- admiral::admiral_adsl
+
+# Convert SAS missing character values to NA
+advs <- vs_metabolic %>%
+  convert_blanks_to_na()
+adsl <- adsl %>%
+  convert_blanks_to_na()
+
+# Merge ADSL variables (stored in `adsl_vars`) needed for ADVS
+advs <- advs %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  )
+
+# Define parameter look-up table used for merging parameter codes to ADVS
+param_lookup <- tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN, ~PARCAT1, ~PARCAT1N,
+  "HEIGHT", "HEIGHT", "Height (cm)", 1, "Anthropometric measurement", 1,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 2, "Anthropometric measurement", 1,
+  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3, "Anthropometric measurement", 1,
```

Comment:
```suggestion
  "BMI", "BMI", "Body Mass Index(kg/m2)", 3, "Anthropometric measurement", 1,
```

![image](https://github.com/user-attachments/assets/9d70f2a0-c70e-4e7d-8520-9dd5d37ef4f5)